controllers/drpolicy_controller_test.go

```diff
@@ -159,8 +161,13 @@ var _ = Describe("DrpolicyController", func() {
 		Expect(k8sClient.Update(context.TODO(), s3Secret)).To(Succeed())
 	}
 	s3SecretUpdateAccessID := func(s3Secret *corev1.Secret, accessID string) {
-		s3Secret.StringData = s3SecretStringData(accessID, s3Secret.StringData["AWS_SECRET_ACCESS_KEY"])
-		s3SecretUpdate(s3Secret)
+		secretFetched := &corev1.Secret{}
+		Expect(k8sClient.Get(
```

Comment:
I wonder how the `s3secret` may not already be the latest from the api server?  And it may not be an issue with envtest, but perhaps this should be `apiReader` client to avoid the cache write bypass issue